src/psyclone/dynamo0p3.py

```diff
@@ -2708,8 +2708,9 @@ def _invoke_declarations(self, parent):
         '''
         api_config = Config.get().api_conf("dynamo0.3")
 
-        # TODO #783: Remove duplicates from the declaration list
         declns = [orient.name+"(:) => null()" for orient in self._orients]
+        # Remove duplicate orientation pointers by using OrderedDict
+        declns = list(OrderedDict.fromkeys(declns))
```

Comment:
I appreciate it is probably subjective but I'm not sure I like the idea of creating duplicates then removing them afterwards. It seems better to create a unique list first then use that i.e. something like ...

```
        unique_orient_names = []
        for orient in self._orients:
            if orient.name not in unique_orient_names:
                unique_orient_names.append(orient.name)
        declns = [name+"(:) => null()" for name in unique_orient_names]
```

Reply:
I chose this way because we have one list iterator (relatively fast) and one `OrderedDict` plus Python built-in list conversion (fast). `OrderedDict`s are one of high-performance container datatypes (e.g. https://docs.python.org/2/library/collections.html) as they are implemented in C from Python 3.5 onwards and they also preserve order. This method was recommended on several Python discussion forums as most efficient without having to `pip` install other specialised packages (e.g. [StacOverflow discussion](https://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-whilst-preserving-order) and a [blog post on list duplicates](https://www.pythonpool.com/python-remove-duplicates-from-list/)).

Reply:
If we are removing duplicates from a list then I am happy with the argument. However, the proposed solution avoids this but not creating duplicates in the first place. I might be missing something but there is only one list iterator in the suggested solution as well and it does not require a conversion to ordereddict and back again to list. I, of course, accept that it takes more lines due to the if test but it is not possible to do via a list comprehension (as far as I know).

Reply:
I've taken a second opinion from @arporter and lost :-) - mainly on the grounds of succinctness, so will accept as is.